src/MenuSectionFormNodeFormAlter.php

```diff
@@ -23,6 +23,7 @@ public static function alter(&$form, NodeInterface $node) {
       if ($available = $nodeType->getThirdPartySetting('menu_ui', 'available_menus', ['main'])) {
         $query->condition('menu_name', array_values($available), 'NOT IN');
       }
+      $query->accessCheck(FALSE);
```

Comment:
**CHANGE REQUEST**

When not provided, access check defaults to `TRUE` so to not change the behavior of the program we should change this to `$query->accessCheck(TRUE);`.

Reply:
https://www.drupal.org/node/3201242

Reply:
I don't know this module well. In the case we might need to work with menu link data even if the user doesn't traditionally have permission to edit it, that was the reason `$query->accessCheck(FALSE);` was used. Are you confident that there isn't a case like that? then, we can enforce accessCheck()

Reply:
This code alters the node form to allow editing a menu link that points to the node, so that you don't have to go find the menu and edit the link there. This query is used to automatically detect the menu. So, yes, we need the access control; if the user does not have access to the link then the user will not be able to edit it anyways, so the menu should not be detected.

Moreover, that's the current behavior and exactly because we don't know the module very well I don't want to change the behavior potentially introducing bugs or security vulnerabilities. This is how it currently works and we just want to port it as is to Drupal 10.

Reply:
```suggestion
      $query->accessCheck(TRUE);
```

Please apply the suggestion and continue the review.